target.go

```diff
@@ -20,35 +20,19 @@
 package multiresolver
 
 import (
-	"strings"
+	"net/url"
 
 	"google.golang.org/grpc/resolver"
 )
 
-// split2 returns the values from strings.SplitN(s, sep, 2).
-// If sep is not found, it returns ("", "", false) instead.
-func split2(s, sep string) (string, string, bool) {
-	spl := strings.SplitN(s, sep, 2)
-	if len(spl) < 2 {
-		return "", "", false
-	}
-	return spl[0], spl[1], true
-}
-
-// ParseTarget splits target into a resolver.Target struct containing scheme,
-// authority and endpoint.
+// ParseTarget parses target into resolver.Target URL
 //
-// If target is not a valid scheme://authority/endpoint, it returns {Endpoint:
-// target}.
+// If target is not a valid URL, it returns {Endpoint: URL: {Path: target}}.
 func ParseTarget(target string) (ret resolver.Target) {
-	var ok bool
-	ret.Scheme, ret.Endpoint, ok = split2(target, "://")
-	if !ok {
-		return resolver.Target{Endpoint: target}
+	u, err := url.Parse(target)
+	if err != nil {
+		return resolver.Target{URL: url.URL{Path: target}}
 	}
-	ret.Authority, ret.Endpoint, ok = split2(ret.Endpoint, "/")
-	if !ok {
-		return resolver.Target{Endpoint: target}
-	}
-	return ret
+
+	return resolver.Target{URL: *u}
```

Comment:
Should we also set .Schema and .Authority? I don't know if anything still reads that (I assume gRPC itself doesn't, but others still might).

Downside is that'll break in a future version of gRPC again.

Reply:
gRPC seem to set deprecated fields, so it might be a good idea to replicate the behavior indeed.